pkg/recycle/drain.go

```diff
@@ -50,7 +51,35 @@ func (r *Recycler) drainNode(helper *drain.Helper) error {
 		return err
 	}
 
-	return drain.RunNodeDrain(helper, r.nodeToRecycle.Name)
+	err = drain.RunNodeDrain(helper, r.nodeToRecycle.Name)
+	if err != nil {
+		pendingList, newErrs := helper.GetPodsForDeletion(r.nodeToRecycle.Name)
+		if pendingList != nil {
```

Comment:
We should test this condition. We could mock a failure and then assert that disableEviction is true.

Reply:
@jasonBirchall , added a e2e test to mock a node with pdb and stuck pod.

Reply:
Test looks great